src/SongList.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from "react";
+import { useHandleFetchAndLoad } from "./useHandleFetchAndLoad";
+
+export const Songlist = (props: { playlistId: string }): React.ReactElement => {
+  const [songList, setSongList] = useState({ index: 0, tracks: [] });
+  const endpoint = `https://api.spotify.com/v1/playlists/${props.playlistId}/tracks?offset=${songList.index}`;
+  var myHeaders = new Headers();
+  myHeaders.append("Authorization", `Bearer ${sessionStorage.accessToken}`);
+
+  var requestOptions = {
+    method: "GET",
+    headers: myHeaders,
+    redirect: "follow",
+  };
+  type fetchData = {
+    items: Array<{ track: {name: string} }>;
+    offset: number;
+    next: string;
+    total: number;
+  };
+  const [loading, data, error] = useHandleFetchAndLoad<fetchData>({
+    endpoint,
+    requestOptions,
+  });
+  console.log(data);
+  if (loading) {
+    return <div>Playlist Loading from {endpoint}</div>;
+  }
+  if (!data){
+    return <div>no data in songlist</div>
+  }
```

Comment:
in general its a good idea to keep your returns the same context.

here you're returning a `div` with content

but below, in the case of it working, you return an unordered list. 

what happens if you contextualize this component with a wrapper? how does that styling differ as you change the return parent of this component? Keeping things coinsistent on this level makes layout way easier down the line